lib/widget/film_stream_widget.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:flutter/material.dart';
+import 'package:my_movies/model/film.dart';
+import 'package:my_movies/widget/film_list_view.dart';
+import 'package:my_movies/widget/film_filter.dart';
+import 'package:my_movies/widget/rating_button.dart';
+
+class FilmStreamWidget extends StatelessWidget {
+  final Stream<List<Film>> filmsStream;
+  final FilmRating? filterRating;
+  final void Function(FilmRating) toggleFilter;
+  final void Function(Film, int?) onSaveFilm;
+  final void Function(int, FilmRating) updateRating;
+
+  const FilmStreamWidget({
+    super.key,
+    required this.filmsStream,
+    required this.filterRating,
+    required this.toggleFilter,
+    required this.onSaveFilm,
+    required this.updateRating,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return StreamBuilder<List<Film>>(
+      stream: filmsStream,
+      builder: (context, snapshot) {
+        if (snapshot.connectionState == ConnectionState.waiting) {
+          return const Center(child: CircularProgressIndicator());
+        } else if (snapshot.hasError) {
+          return Center(child: Text('Error: ${snapshot.error}'));
+        } else if (!snapshot.hasData || snapshot.data!.isEmpty) {
+          return const Center(child: Text('Add your first film!'));
+        } else {
```

Comment:
Pokud mám if (...) { return }, tak pak už není potřeba else větev.

Pro StreamBuilder si stačí ověřit jen:
- if (snapshot.hasError)
- if (!snapshot.hasData) - čekám na data ze streamy (např. než mi databáze vrátí výsledek)

A potom už zobrazovat content (alternativně pokud je ten list empty, tak zobrazit nějakou hlášku místo prázdné stránky)